src/IsolatedTestHost/Program.cs

```diff
@@ -0,0 +1,117 @@
+﻿namespace IsolatedTestHost
+{
+    using System;
+    using System.IO;
+    using System.Linq;
+    using System.Reflection;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Xunit;
+
+    internal static class Program
+    {
+        private static int Main(string[] args)
+        {
+            if (args.Length < 3)
+            {
+                return (int)ExitCodes.CommandLineArgsIncomplete;
+            }
+
+            string assemblyFile = args[0];
```

Comment:
I think using deconstruction where it makes sense is good. But... IMO the syntax you propose makes it harder to see the index for each argument.